### ISSUE TITLE
PublishEvent  option failed to set keys TTL

### ISSUE THREAD

**Issue by MrGuoRanDuo** [dapr/go-sdk#385]
**Describe the bug**
In use the PublishEvent function , I set the PublishEventOption : 
`client.PublishEventWithMetadata(map[string]string{"ttlInseconds":"3600"})`  
But when I check redis :
127.0.0.1:6379> TTL Stop
(integer) -1
clearly failed to set ttl , the value is -1  , but key store success.
And I don't see anything ttl unit test in client/pubsub_test.go

**To Reproduce**
redis docker version 6.2.5
go version 1.19
github.com/dapr/dapr v1.10.4
github.com/dapr/go-sdk v1.7.0

Hear's full codes: 
`if err := d.daprClient.PublishEvent(ctx, "redis-pubsub", master.MethodType_Stop.String(), param,
		client.PublishEventWithMetadata(map[string]string{"ttlInseconds": "3600"})); err != nil {
		return err
	}`

**Expected behavior**
1. If this is user mistake, please point where I was wrong.
2. fix this bug, pls

**Comment by Taction** [comment]
For those components that don't have built-in ttl support, dapr will handle it in runtime. This means dapr will check the TTL and ignore these expired messages.

**Comment by MrGuoRanDuo** [comment]
@Taction Thx for your reply
In this case, Redis store too much msg , i have to delete them manually(because I lost one msg, may cause by redis perforemance issue).
My question ：Msgs are not delete after they are receviced(OnTopicEvent), is that right?

**Comment by Taction** [comment]
> My question ：Msgs are not delete after they are receviced(OnTopicEvent), is that right?
Yes, Dapr redis pubsub component uses redis stream which doesn't delete messages when acked.

For your situation, you can set `maxLenApprox` metadata, which defines maximum number of items inside a stream. The old entries are automatically evicted when the specified length is reached so that the stream will be left at a constant size. Or you can try other components (such as rabbitmq), which delete the expired or acked message.

**Comment by wXwcoder** [comment]
> **Describe the bug** In use the PublishEvent function , I set the PublishEventOption : `client.PublishEventWithMetadata(map[string]string{"ttlInseconds":"3600"})` But when I check redis : 127.0.0.1:6379> TTL Stop (integer) -1 clearly failed to set ttl , the value is -1 , but key store success. And I don't see anything ttl unit test in client/pubsub_test.go
> 
> **To Reproduce** redis docker version 6.2.5 go version 1.19 github.com/dapr/dapr v1.10.4 github.com/dapr/go-sdk v1.7.0
> 
> Hear's full codes: `if err := d.daprClient.PublishEvent(ctx, "redis-pubsub", master.MethodType_Stop.String(), param, client.PublishEventWithMetadata(map[string]string{"ttlInseconds": "3600"})); err != nil { return err }`
> 
> **Expected behavior**
> 
> 1. If this is user mistake, please point where I was wrong.
> 2. fix this bug, pls

Try modifying ttlInseconds to ttlInSeconds

**Comment by zarkodam** [comment]
I have the same problem with C# client, when using pubsub with Redis streams:

var metadata = new Dictionary<string, string>
            {
                {"ttlInSeconds", "120"}
            };

            await _daprClient.PublishEventAsync(
                pubsubName: "pubsub",
                topicName: "test",
                data: message,
                metadata: metadata, 
                cancellationToken: cancellationToken);

but when I'm using state store with Redis it works well:

await _daprClient.SaveStateAsync("statestore", "test", 1, metadata: metadata);
